### DISCUSSION TITLE
Hyperlinks

### DISCUSSION THREAD

**Discussion by GalaxyPony33** [seleniumbase/SeleniumBase#2239] [Q&A] [answered]
Hi so I have a few questions: (I should of put a better title for this discussion)

1. So what if there's a button on a website I need to click and if I do click it, it will open a link in a new tab and bring focus to the new tab automatically . Is there a chance the bot can be detected in the new tab since I didn't use driver methods to open the website myself? If so, how do I combat this?
2. Let's say we type the url ourselves with keyboard instead of driver methods to go to a website, this method is not recommended right? because I tried going to https://nowsecure.nl/#relax manually on SB with UC and it failed captcha. What can I do to still be able to use keyboard myself and go to websites and not get hit if I prefer no driver methods.
3. We disconnect then reconnect if the bot detects the website going to has anti-bot services, after we bypasses things like the cloudflare captcha wouldn't they be able to try and check us again at a random time and detect that we are bot? It seems like we need to hide chromedriver with the disconnect and reconnect part before every interaction such as loading up a website, clicking something on the website, etc. If we only hide when we are about to interact or do something, what about the times chromedriver is exposed to to the website even when we are not about to do something or interact? So I am little confused on this.

Thank you very much!

**Comment by mdmintz** [comment, marked as answer]
**One:** SeleniumBase automatically opens links in new tabs if there's bot-detection software on those websites. (It checks first). There are additional methods for that: (Covered in this video: https://www.youtube.com/watch?v=5dMFI3e85ig)

```python
driver.uc_open(url)
driver.uc_open_with_tab(url)
driver.uc_open_with_reconnect(url, reconnect_time=None)
driver.uc_click(selector)
```

Be sure to use those methods when the default behavior isn't good enough. And be sure to watch the video for details.

**Two:** If you directly type a URL with the keyboard into the browser's address bar, then you're going to get detected because chromedriver is still attached to Chrome. That's the primary thing that bot-detection services check for. SeleniumBase UC Mode disconnects chromedriver from Chrome briefly when loading a website that has anti-bot services running on it. This tricks those websites into thinking that web automation is not involved, and therefore they believe the browser is being fully driven by a human. If you need the flexibility to issue commands manually, then drop in a Python breakpoint, and run commands from there:

```python
import pdb; pdb.set_trace()  # Or "breakpoint()"
```

Type `c` + `Return` to continue from the breakpoint.

**Three:** Most cloudflare checks only happen on the initial page load. And when they don't (eg. filling out a form and clicking), then disconnecting chromedriver from Chrome right after a click is often fast enough for Selenium to hide itself before the website can detect it. If cloudflare performed random checks (or was continuously looking), then it would be a problem, but so far, it appears they're only looking at specific times (first load of a website, or clicking a specific button). This, of course, is not a perfect system when you have to do more than just get through the front door of a website. (Again, things covered in https://www.youtube.com/watch?v=5dMFI3e85ig)
